sql/2024/sustainability/green_third_party_requests.sql

```diff
@@ -0,0 +1,137 @@
+#standardSQL
+# Median number of third-parties & green third-party requests per websites by rank
+
+WITH requests AS (
+  SELECT
+    client,
+    CAST(JSON_VALUE(summary, '$.pageid') AS INT64) AS page,
+    url
+  FROM
+    `httparchive.all.requests`
+  WHERE
+    date = '2024-06-01'
+),
+
+green AS (
+  SELECT
+    NET.HOST(url) AS host,
+    TRUE AS is_green
+  FROM
+    `httparchive.almanac.green_web_foundation`
+  WHERE
+    date = '2024-09-01'
+),
+
+pages AS (
+  SELECT
+    client,
+    CAST(JSON_VALUE(summary, '$.pageid') AS INT64)AS page,
+    rank
+  FROM
+    `httparchive.all.pages`
+  WHERE
+    date = '2024-06-01'
+),
+
+third_party AS (
+  SELECT
+    domain,
+    COUNT(DISTINCT page) AS page_usage
+  FROM
+    `httparchive.almanac.third_parties` tp
+  JOIN
+    requests r
+  ON NET.HOST(r.url) = NET.HOST(tp.domain)
+  WHERE
+    date = '2024-07-01' AND
+    category NOT IN ('hosting')
+  GROUP BY
+    domain
+  HAVING
+    page_usage >= 50
+),
+
+green_tp AS (
+  SELECT
+    domain
+  FROM
+    `httparchive.almanac.third_parties` tp
+  JOIN
+    green g
+  ON NET.HOST(g.host) = NET.HOST(tp.domain)
+  WHERE
+    date = '2024-07-01' AND
```

Comment:
Dataset is the same but let's be consistent.

```suggestion
    date = '2024-06-01' AND
    category NOT IN ('hosting')
  GROUP BY
    domain
  HAVING
    page_usage >= 50
),

green_tp AS (
  SELECT
    domain
  FROM
    `httparchive.almanac.third_parties` tp
  JOIN
    green g
  ON NET.HOST(g.host) = NET.HOST(tp.domain)
  WHERE
    date = '2024-06-01' AND
```